include/morphio/enums.h

```diff
@@ -60,8 +60,16 @@ enum SectionType {
     SECTION_DENDRITE = 3,         //!< general or basal dendrite (near to soma)
     SECTION_APICAL_DENDRITE = 4,  //!< apical dendrite (far from soma)
 
-    SECTION_GLIA_PERIVASCULAR_PROCESS = 2,
-    SECTION_GLIA_PROCESS = 3,  // TODO: nasty overload there
+    SECTION_GLIA_PERIVASCULAR_PROCESS = 2,  // Note: overlaps with SECTION_AXON
+    SECTION_GLIA_PROCESS = 3,               // Note: overlaps with SECTION_DENDRITE
+
+    SECTION_SPINE_HEAD = 2,  // Note: overlaps with SECTION_AXON
```

Comment:
The HDF5 Morphology [specification]( https://bbpteam.epfl.ch/documentation/projects/Morphology%20Documentation/latest/h5v1.html) defines the type of process for dendrite spine section as follow:
1: head, 2: neck

Is there a reason why you defined other values ?

Reply:
Yeah, it's a work around for now so you can test the API; the overlap w/ the soma and the different handling of it.  It will go back to the 'original' values once I have a clean workaround for it.

Reply:
@jacquemi-bbp thinking about the numbering here, I wonder if it makes sense to keep the 'work around'; does the `head` behave like a `soma`?  Is there only a single one of them, for instance?  In the documentation, they seem to be more common, and actually attach to different necks

Reply:
There can be several heads, attached to the same "neck" section.

Reply:
Yeah, that's why I think that it's not really comparable to a `soma` and why using a different constant makes sense.

Reply:
Furthermore, I made a mistake on the documentation example. The green section is neck type (2) and the red one is head type (1) as the blue one

Reply:
Could you update the documentation with the new IDs, and also the changes needed to fix the mistake?

Reply:
https://bbpcode.epfl.ch/code/#/c/53854/